staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/metrics/metrics_test.go

```diff
@@ -0,0 +1,139 @@
+        	apiserver_envelope_encryption_key_id_hash_total{key_id_hash="%s",provider_name="providerName",transformation_type="from_storage"} 1
+        	apiserver_envelope_encryption_key_id_hash_total{key_id_hash="%s",provider_name="providerName",transformation_type="from_storage"} 1
+			`, testKeyHash1, testKeyHash2),
+		},
+		{
+			desc:  "keyIDHash total same labels",
+			keyID: "2",
+			metrics: []string{
+				"apiserver_envelope_encryption_key_id_hash_total",
+			},
+			providerName:       testProviderName,
+			transformationType: "from_storage",
+			want: fmt.Sprintf(`
+			# HELP apiserver_envelope_encryption_key_id_hash_total [ALPHA] Number of times a keyID is used split by transformation type and provider.
+			# TYPE apiserver_envelope_encryption_key_id_hash_total counter
+			apiserver_envelope_encryption_key_id_hash_total{key_id_hash="%s",provider_name="providerName",transformation_type="from_storage"} 1
+			apiserver_envelope_encryption_key_id_hash_total{key_id_hash="%s",provider_name="providerName",transformation_type="from_storage"} 2
+			`, testKeyHash1, testKeyHash2),
+		},
+		{
+			desc:  "keyIDHash total exceeds limit, remove first label",
+			keyID: "3",
+			metrics: []string{
+				"apiserver_envelope_encryption_key_id_hash_total",
+			},
+			providerName:       testProviderName,
+			transformationType: "from_storage",
+			want: fmt.Sprintf(`
+			# HELP apiserver_envelope_encryption_key_id_hash_total [ALPHA] Number of times a keyID is used split by transformation type and provider.
+			# TYPE apiserver_envelope_encryption_key_id_hash_total counter
+			apiserver_envelope_encryption_key_id_hash_total{key_id_hash="%s",provider_name="providerName",transformation_type="from_storage"} 2
+			apiserver_envelope_encryption_key_id_hash_total{key_id_hash="%s",provider_name="providerName",transformation_type="from_storage"} 1
+			`, testKeyHash2, testKeyHash3),
+		},
+		{
+			desc:  "keyIDHash total exceeds limit 2, remove first label",
+			keyID: "1",
+			metrics: []string{
+				"apiserver_envelope_encryption_key_id_hash_total",
+			},
+			providerName:       testProviderName,
+			transformationType: "from_storage",
+			want: fmt.Sprintf(`
+			# HELP apiserver_envelope_encryption_key_id_hash_total [ALPHA] Number of times a keyID is used split by transformation type and provider.
+			# TYPE apiserver_envelope_encryption_key_id_hash_total counter
+			apiserver_envelope_encryption_key_id_hash_total{key_id_hash="%s",provider_name="providerName",transformation_type="from_storage"} 1
+			apiserver_envelope_encryption_key_id_hash_total{key_id_hash="%s",provider_name="providerName",transformation_type="from_storage"} 1
+			`, testKeyHash3, testKeyHash1),
+		},
+	}
+
+	RegisterMetrics()
+	KeyIDHashTotal.Reset()
+
+	for _, tt := range testCases {
+		t.Run(tt.desc, func(t *testing.T) {
+			maxKeyIDLabelsLimit = 2
+			RecordKeyID(tt.transformationType, tt.providerName, tt.keyID)
+			//defer KeyIDHashTotal.Reset()
```

Comment:
?

Does this need a comment about how each test is not independent and builds on the one that came before it?